src/components/DeeplinkWrapper/index.website.js

```diff
@@ -1,41 +1,90 @@
+    // 2. There may be non-idempotent operations (e.g. create a new workspace), which obviously should not be executed again in the desktop app.
+    // So we need to wait until after sign-in and navigation are complete before starting the deeplink redirect.
+    if (Str.startsWith(window.location.pathname, Str.normalizeUrl(ROUTES.TRANSITION_BETWEEN_APPS))) {
+        App.beginDeepLinkRedirectAfterTransition();
+    } else {
+        App.beginDeepLinkRedirect();
+    }
+}
+function DeeplinkWrapper({children, isAuthenticated}) {
+    const [currentScreen, setCurrentScreen] = useState();
+    const [hasShownPrompt, setHasShownPrompt] = useState(false);
+    const removeListener = useRef();
+
+    useEffect(() => {
+        // If we've shown the prompt and still have a listener registered,
+        // remove the listener and reset its ref to undefined
+        if (hasShownPrompt && removeListener.current !== undefined) {
+            removeListener.current();
+            removeListener.current = undefined;
         }
 
-        // If the current url path is /transition..., meaning it was opened from oldDot, during this transition period:
-        // 1. The user session may not exist, because sign-in has not been completed yet.
-        // 2. There may be non-idempotent operations (e.g. create a new workspace), which obviously should not be executed again in the desktop app.
-        // So we need to wait until after sign-in and navigation are complete before starting the deeplink redirect.
-        if (Str.startsWith(window.location.pathname, Str.normalizeUrl(ROUTES.TRANSITION_BETWEEN_APPS))) {
-            App.beginDeepLinkRedirectAfterTransition();
+        if (isAuthenticated === false) {
+            setHasShownPrompt(false);
+            Navigation.isNavigationReady().then(() => {
+                // get initial route
+                const initialRoute = navigationRef.current.getCurrentRoute();
+                setCurrentScreen(initialRoute.name);
+
+                removeListener.current = navigationRef.current.addListener('state', (event) => {
+                    // accessing routes here should be in a navigation lib fn in case the state shape changes in the future
+                    setCurrentScreen(event.data.state.routes.slice(-1).name);
+                });
+            });
+        }
+    }, [hasShownPrompt, isAuthenticated]);
+    useEffect(() => {
+        // Making a few checks to exit early before checking authentication status
+        if (!isMacOSWeb() || CONFIG.ENVIRONMENT === CONST.ENVIRONMENT.DEV || hasShownPrompt) {
             return;
         }
-        App.beginDeepLinkRedirect();
-    }
 
-    isMacOSWeb() {
-        return !Browser.isMobile() && typeof navigator === 'object' && typeof navigator.userAgent === 'string' && /Mac/i.test(navigator.userAgent) && !/Electron/i.test(navigator.userAgent);
-    }
+        if (isAuthenticated) {
+            promptToOpenInDesktopApp();
+            setHasShownPrompt(true);
```

Comment:
This is firing twice on authenticated routes for me. I think in the refactor the use of `hasShownPrompt` to make sure we don't show it multiple times got removed. I think `hasShownPrompt` as a return guard before the `if (isAuthenticated)` would work, right?

Reply:
🤦‍♀️ My mistake, I removed that line to test and I hadn't noticed it changed. All good then 👍